pkgs/tools/text/gawk/default.nix

```diff
@@ -74,9 +45,66 @@ stdenv.mkDerivation rec {
 
     license = licenses.gpl3Plus;
 
-    platforms = platforms.unix;
+    platforms = platforms.unix ++ platforms.windows;
 
     maintainers = [ ];
   };
-}
+in
+
+if stdenv.hostPlatform.isWindows then
+  stdenv.mkDerivation rec {
+    inherit src name meta;
+
+    # $src/pc has pre-configured makefiles for Windows
+    configurePhase = ''
+      substituteInPlace pc/gawkmisc.pc    --replace 'int execvp(const char *file, const char *const *argv)' 'int execvp(const char *file, char *const *argv)'
+      substituteInPlace pc/Makefile       --replace CC=gcc               CC=${stdenv.cc.targetPrefix}cc
+      substituteInPlace pc/Makefile       --replace 'prefix = c:/gnu'    "prefix = $out"
+      substituteInPlace pc/Makefile.ext   --replace 'prefix = c:/gnu'    "prefix = $out"
```

Comment:
Should it not possible to override prefix and CC by passing it to make instead of patching?